node/hack/repair_terra/repair.go

```diff
@@ -0,0 +1,369 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/db"
+	"go.uber.org/zap"
+
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	nodev1 "github.com/certusone/wormhole/node/pkg/proto/node/v1"
+	"github.com/certusone/wormhole/node/pkg/vaa"
+	eth_common "github.com/ethereum/go-ethereum/common"
+
+	"github.com/tidwall/gjson"
+	"google.golang.org/grpc"
+)
+
+var (
+	adminRPC   = flag.String("adminRPC", "/run/guardiand/admin.socket", "Admin RPC address")
+	terraAddr  = flag.String("terraProgram", "terra1dq03ugtd40zu9hcgdzrsq6z2z4hwhc9tqk2uy5", "Terra program address")
+	dryRun     = flag.Bool("dryRun", true, "Dry run")
+)
+
+func getAdminClient(ctx context.Context, addr string) (*grpc.ClientConn, error, nodev1.NodePrivilegedServiceClient) {
+	conn, err := grpc.DialContext(ctx, fmt.Sprintf("unix:///%s", addr), grpc.WithInsecure())
+
+	if err != nil {
+		log.Fatalf("failed to connect to %s: %v", addr, err)
+	}
+
+	c := nodev1.NewNodePrivilegedServiceClient(conn)
+	return conn, err, c
+}
+
+func getSequenceForTxhash(txhash string) (uint64, error) {
+	client := &http.Client{ 
+		Timeout: time.Second * 5, 
+	}
+	url := fmt.Sprintf("https://fcd.terra.dev/cosmos/tx/v1beta1/txs/%s", txhash)
+	resp, err := client.Get(url)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get message: %w", err)
+	}
+	defer resp.Body.Close()
+	txBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		resp.Body.Close()
+		return 0, fmt.Errorf("failed to read message: %w", err)
+	}
+
+	txJSON := string(txBody)
+	if !gjson.Valid(txJSON) {
+		return 0, fmt.Errorf("invalid JSON response")
+	}
+	txHashRaw := gjson.Get(txJSON, "tx_response.txhash")
+	if !txHashRaw.Exists() {
+		return 0, fmt.Errorf("terra tx does not have tx hash")
+	}
+	txHash := txHashRaw.String()
+
+	events := gjson.Get(txJSON, "tx_response.events")
+	if !events.Exists() {
+		return 0, fmt.Errorf("terra tx has no events")
+	}
+	msgs := EventsToMessagePublications(*terraAddr, txHash, events.Array())
+	// Should only ever be 1 message. Stole the above function from watcher.go
+	if len(msgs) != 1 {
+		return 0, fmt.Errorf("EventsToMessagePublications returned %d msgs", len(msgs))
+	}
+	return msgs[0].Sequence, nil
+}
+
+// This was stolen from pkg/terra/watcher.go
+func EventsToMessagePublications(contract string, txHash string, events []gjson.Result) []*common.MessagePublication {
```

Comment:
Why not import the real function? Logging?

Reply:
The original function was logging way too much.